api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupTPTInitiationReport.java

```diff
@@ -0,0 +1,100 @@
+package org.openmrs.module.eptsreports.reporting.reports;
+
+import java.io.IOException;
+import java.util.*;
+import org.openmrs.Location;
+import org.openmrs.module.eptsreports.reporting.library.cohorts.GenericCohortQueries;
+import org.openmrs.module.eptsreports.reporting.library.datasets.TPTInitiationDataset;
+import org.openmrs.module.eptsreports.reporting.reports.manager.EptsDataExportManager;
+import org.openmrs.module.eptsreports.reporting.utils.EptsReportUtils;
+import org.openmrs.module.reporting.ReportingException;
+import org.openmrs.module.reporting.evaluation.parameter.Mapped;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.openmrs.module.reporting.report.ReportDesign;
+import org.openmrs.module.reporting.report.definition.ReportDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SetupTPTInitiationReport extends EptsDataExportManager {
+
+  private TPTInitiationDataset tptInitiationDataset;
+
+  @Autowired protected GenericCohortQueries genericCohortQueries;
+
+  @Autowired
+  public SetupTPTInitiationReport(TPTInitiationDataset tptInitiationDataset) {
+    this.tptInitiationDataset = tptInitiationDataset;
+  }
+
+  @Override
+  public String getExcelDesignUuid() {
+    return "0638ab60-bf2c-11e9-b8ed-7b0ec2ec93ad";
+  }
+
+  @Override
+  public String getUuid() {
+    return "10b6239c-bf2c-11e9-a2e5-63fb38259292";
+  }
+
+  @Override
+  public String getName() {
+    return "TPT Initiation Report";
+  }
+
+  @Override
+  public String getDescription() {
+    return "This report provides a line listing of patient records who initiated TPT";
+  }
+
+  @Override
+  public ReportDefinition constructReportDefinition() {
+    ReportDefinition rd = new ReportDefinition();
+    rd.setUuid(getUuid());
+    rd.setName(getName());
+    rd.setDescription(getDescription());
+    rd.addParameters(getParameters());
+
+    rd.setBaseCohortDefinition(
+        EptsReportUtils.map(
+            genericCohortQueries.getBaseCohort(),
+            "startDate=${startDate}, endDate=${endDate},location=${location}"));
+    rd.addDataSetDefinition(
+        "ALL", Mapped.mapStraightThrough(tptInitiationDataset.constructDatset(getParameters())));
```

Comment:
@Edgencio any reason to pass the parameters from here instead of using the one you already extended from the dataset?